test/conformance/api/v1/service_test.go

```diff
@@ -169,7 +169,7 @@ func waitForDesiredTrafficShape(t *testing.T, sName string, want map[string]v1.T
 			// Match the traffic shape.
 			got := map[string]v1.TrafficTarget{}
 			for _, tt := range s.Status.Traffic {
-				got[tt.Tag] = tt
+				got[tt.Tag] = setTrafficTargetDefaults(tt)
```

Comment:
I think we could theoretically do this with something like the following to be consistent with how we ignore URLs, but maybe (almost certainly 😅) Im just being too clever by half with this. I'm OK if we don't bother, just throwing the thought out there in case you prefer this way.

~~~~
ignoreURLs := cmpopts.IgnoreFields(v1.TrafficTarget{}, "URL") // existing line
allowNilDefaults := cmp.Transformer("AllowNilDefaults", setTrafficTargetDefaults)
if !cmp.Equal(got, want, ignoreURLs, allowNilDefaults) {
~~~~

Reply:
`The transformer f must be a function "func(T) R" that converts values of type T to those of type R and is implicitly filtered to input values assignable to T. The transformer must not mutate T in any way.`

We are mutating the original value in `setTrafficTargetDefaults` and I think it is ok. I'd just keep as it is.

Reply:
I don't think we'll be mutating it because it not a pointer (and we could copy it to avoid mutation if it were), but it doesn't really matter, I'm fine with the current way